examples/hotel_ads/add_performance_max_for_travel_goals_campaign.rb

```diff
@@ -0,0 +1,823 @@
+        # the mutate request that should have its status set.
+        c.status = :PAUSED
+
+        # All Performance Max campaigns have an advertising_channel_type of
+        # PERFORMANCE_MAX. The advertising_channel_sub_type should not be set.
+        c.advertising_channel_type = :PERFORMANCE_MAX
+
+        # To create a Performance Max for travel goals campaign, you need to set hotel_property_asset_set
+        c.hotel_property_asset_set = hotel_property_asset_set_resource_name
+
+        # Bidding strategy must be set directly on the campaign.
+        # Setting a portfolio bidding strategy by resource name is not supported.
+        # Max Conversion and Maximize Conversion Value are the only strategies
+        # supported for Performance Max campaigns.
+        # An optional ROAS (Return on Advertising Spend) can be set for
+        # maximize_conversion_value. The ROAS value must be specified as a ratio
+        # in the API. It is calculated by dividing "total value" by "total spend".
+        # For more information on Maximize Conversion Value, see the support
+        # article: http://support.google.com/google-ads/answer/7684216.
+        # A target_roas of 3.5 corresponds to a 350% return on ad spend.
+        c.bidding_strategy_type = :MAXIMIZE_CONVERSION_VALUE
+        c.maximize_conversion_value =
+          client.resource.maximize_conversion_value do |mcv|
+            mcv.target_roas = 3.5
+          end
+
+        # Assign the resource name with a temporary ID.
+        c.resource_name =
+          client.path.campaign(
+            customer_id,
+            PERFORMANCE_MAX_CAMPAIGN_TEMPORARY_ID
+          )
+
+        # Set the budget using the given budget resource name.
+        c.campaign_budget =
+          client.path.campaign_budget(customer_id, BUDGET_TEMPORARY_ID)
+      end
+  end
+end
+# [END create_campaign]
+
+def create_campaign_budget_operation(client, customer_id)
+  client.operation.mutate do |m|
+    m.campaign_budget_operation =
+      client.operation.create_resource.campaign_budget do |cb|
+        cb.name = "Performance Max campaign budget #{SecureRandom.uuid}"
+        # The budget period already defaults to DAILY.
+        cb.amount_micros = 50_000_000
+        cb.delivery_method = :STANDARD
+        # A Performance Max campaign cannot use a shared campaign budget.
+        cb.explicitly_shared = false
+
+        # Set a temporary ID in the budget's resource name so it can be referenced
+        # by the campaign in later steps.
+        cb.resource_name =
+          client.path.campaign_budget(customer_id, BUDGET_TEMPORARY_ID)
+      end
+  end
+end
+
+# [START get_hotel_asset_suggestion]
+def get_hotel_asset_suggestion(client, customer_id, place_id)
+  response =
+    client.service.travel_asset_suggestion.suggest_travel_assets(
+      customer_id: customer_id,
+      language_option: 'en-US',
+      place_id: [place_id]
+    )
+
+  response.hotel_asset_suggestions[0]
```

Comment:
We usually use `.first` in ruby examples for this.

Reply:
done